src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -497,19 +488,51 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                         Ok((res, Some(format!("{}.{}", kind, item_str))))
                     }
                 }),
+            Res::PrimTy(prim) => {
```

Comment:
You might want to handle the case where primitives get resolved to `Res::Mod`. That was part of the cleanup PR but was removed when the behaviour change was reverted.

Reply:
This change was part of a larger refactor that I don't want to make right now, since otherwise this will never land.